contracts/reverseRegistrar/profiles/L2NameResolver.sol

```diff
@@ -0,0 +1,40 @@
+// SPDX-License-Identifier: MIT
+pragma solidity >=0.8.4;
+
+import "./L2ReverseResolverBase.sol";
+import "../../resolvers/profiles/INameResolver.sol";
+
+abstract contract L2NameResolver is INameResolver, L2ReverseResolverBase {
```

Comment:
Is this different to `NameResolver`? Can't we avoid the need for duplication here by implementing the L2-specific functionality as a mixin at the top of the inheritance hierarchy, instead of the bottom?

Reply:
So the problem I had was, I needed access to the address not the node in authorisation. And the original resolver uses `bytes32 node`. Whereas the reverse registrar passes through address for authorisation. I wasn't sure how to change the modifier here without duplicating the contracts.